src/bank.rs

```diff
@@ -206,45 +317,70 @@ impl Bank {
         let bank = Self::default();
         {
             let mut accounts = bank.accounts.write().unwrap();
-            let account = accounts.entry(deposit.to).or_insert_with(Account::default);
-            Self::apply_payment(deposit, account);
+
+            let mut account = Account::default();
+            account.tokens += deposit.tokens;
+
+            accounts.store(&deposit.to, &account);
         }
         bank.add_builtin_programs();
         bank
     }
 
+    pub fn checkpoint(&self) {
+        self.accounts.write().unwrap().checkpoint();
+        self.last_ids.write().unwrap().checkpoint();
+    }
+    pub fn rollback(&self) {
+        let rolled_back_pubkeys: Vec<Pubkey> = self
+            .accounts
+            .read()
+            .unwrap()
+            .accounts
+            .keys()
+            .map(|pubkey| pubkey.clone())
+            .collect();
+
+        self.accounts.write().unwrap().rollback();
```

Comment:
Will there be  issues  if there are ongoing transactions being processed in execute_and_commit_transactions(), which loads the account, operates on it, and then stores it at some arbitrary point in the future?